### PR TITLE
LPD-20696  Admin Return Workflow - part 1

### PR DESCRIPTION
**Pull request by ghost**
Sending json files, validators and constant files

https://liferay.atlassian.net/browse/LPD-20696

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/00-list-type-definition.batch-engine-data.json [hunk 1]
```diff
@@ -14,6 +14,104 @@
 		"version": "v1.0"
 	},
 	"items": [
+		{
+			"externalReferenceCode": "L_COMMERCE_RETURN_ITEM_STATUSES",
```
Comment (smottal) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/00-list-type-definition.batch-engine-data.json:18]: Are all these status managed in the typescript component that print the "label"?

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/03-object-definition.batch-engine-data.json [hunk 2]
```diff
@@ -83,6 +83,38 @@
 					"system": true,
 					"type": "BigDecimal"
 				},
+				{
+					"DBType": "Long",
```
Comment (smottal) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/03-object-definition.batch-engine-data.json:87]: I think all the "quantities" this should be BigDecimal

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json [hunk 3]
```diff
@@ -68,56 +68,28 @@
...
-					"required": true,
-					"state": true,
-					"system": true,
-					"type": "String"
-				},
 				{
 					"DBType": "String",
 					"businessType": "Aggregation",
-					"externalReferenceCode": "RETURN_REQUESTED_ITEMS",
+					"externalReferenceCode": "RETURN_CHANNEL_TOTAL_ACCEPTED",
```
Comment (smottal) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:74]: Mmm... we are summarizing different things (liter accepted + meters accepted). Are we sure?

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json [hunk 4]
```diff
@@ -68,56 +68,28 @@
...
 					"indexed": false,
 					"indexedAsKeyword": false,
 					"label": {
-						"en_US": "Requested Items"
+						"en_US": "Total Accepted"
 					},
-					"name": "requestedItems",
+					"name": "totalReceived",
```
Comment (smottal) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:80]: Mmm... we are summarizing different things (liter received + meters received). Are we sure?

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json [hunk 5]
```diff
@@ -157,13 +129,13 @@
 				{
 					"DBType": "String",
 					"businessType": "Aggregation",
-					"externalReferenceCode": "RETURN_CHANNEL_TOTAL_AMOUNT",
+					"externalReferenceCode": "RETURN_CHANNEL_TOTAL_AUTHORIZED",
```
Comment (smottal) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:132]: Mmm... we are summarizing different things (liter authorized + meters authorized). Are we sure?

**Review by smottal — CHANGES_REQUESTED** [review]
@CrescenzoRegaLF just an idea: what about creating an abstract CommerceReturnObjectValidationRuleEngine that contains some methods to retrieve common data? Because all of the validations does more or less the same things (like getting the `entryDTO`, then the `properties` and from the properties the values)

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json (hunk 3 above)
Comment (smottal) [reply modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:74]: why RETURN_CHANNEL_TOTAL_ACCEPTED? I mean... why CHANNEL?

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json [hunk 6]
```diff
@@ -128,13 +100,13 @@
 				{
 					"DBType": "String",
 					"businessType": "Aggregation",
-					"externalReferenceCode": "RETURN_CHANNEL_TOTAL_ACCEPTED",
+					"externalReferenceCode": "RETURN_CHANNEL_TOTAL_AMOUNT",
```
Comment (smottal) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:103]: why RETURN_CHANNEL_TOTAL_AMOUNT? I mean... why CHANNEL?

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json (hunk 5 above)
Comment (smottal) [reply modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:132]: why RETURN_CHANNEL_TOTAL_AUTHORIZED? I mean... why CHANNEL?

**Review by smottal — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json (hunk 4 above)
Comment (smottal) [reply modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:80]: why the name is totalReceived if is total_accepted?

**Review by smottal — CHANGES_REQUESTED** [review]

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json (hunk 3 above)
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:74]: they are names that if I remember correctly you actually chose them

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json (hunk 5 above)
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:132]: they are names that if I remember correctly you actually chose them

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json (hunk 6 above)
Comment (CrescenzoRegaLF) [reply modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/04-object-definition.batch-engine-data.json:103]: they are names that if I remember correctly you actually chose them

**Pull request closed without merging**